.github/workflows/reusable_pull_request.yml

```diff
@@ -83,4 +91,39 @@ jobs:
 
         for target in "${targets[@]}"; do
           swift package plugin generate-documentation --target "$target" --warnings-as-errors --analyze --level detailed
-        done
\ No newline at end of file
+        done
+
+  acceptable-language-check:
+    name: Acceptable language check
+    if: ${{ inputs.acceptable_language_check_enabled }}
+    runs-on: ubuntu-latest
+    timeout-minutes: 1
+    steps:
+    - name: Checkout repository
+      uses: actions/checkout@v4
+    - name: Run acceptable language check
+      shell: bash
+      env:
+        UNACCEPTABLE_LANGUAGE_PATTERNS: ${{ inputs.acceptable_language_check_word_list}}
+      run: |
+        set -euo pipefail
```

Comment:
could we put scripts in actual files and run `shellcheck` over them? It's extra error-prone to program in a `.yml` file. If we just have a `dev/action_check_unacceptable_language.sh` or something, then you can just invoke that here.

That makes it easier to run `shellcheck` too

Reply:
The reason why these script are not in a file is that we want to move this reusable workflow out of the NIO repo and share it across all of our package repos for easier maintenance. Shared workflows can't pull in any files from their repo sadly. So we have to make the scripts inline in the yaml. That's not great but IMO worth the trade of copying scripts around

Reply:
I also initially wanted to have the scripts locally but I think the goal here is that we don't have to copy them into every repo long term. And yes, I use shellcheck and, in the OpenAPI Generator project, we use `shellcheck` in CI on the CI scripts themselves.

Reply:
I get the issue that we want shared resources without having to copy stuff around. I don't think embedding them in `.yml` is a great solution for this issue.

What about creating a `swift-nio-ci-helpers` or something repo that we just `git clone` or have available too?  I know it's not great either but there has to be a better way than `yml` embedding. That makes testing the scripts locally hard (which leads to bugs), that makes shellcheck'ing them hard (leading to more bugs) and requires users to learn the weird yml syntax (which leads to frustration -> more bugs still)

Reply:
We already have a solution for testing locally which I have been using and it works great. It's called [act](https://nektosact.com/introduction.html) and allows you run the full workflows locally. It reduces the overhead of scripts a lot and contributors can just do `act pull_request` and run their whole CI checks

Reply:
Definitely an increase in faff, but we could separate the scripts out and generate the yaml files with another script (basically just to inline the scripts into the yaml). We could also shellcheck the scripts in CI + check the yaml is up-to-date.

Reply:
I agree though that having the scripts inline isn't great and I want to change that in the future once we have everything in place and move the reusable workflow into its final place actually

Reply:
Please, this requires users to install more software. That adds friction and frustration.

Let's take a step back: Embedding multiple files into one `.yml` file just because we can't figure out how to share code across a set of repos seems really wrong.

Reply:
> having the scripts inline isn't great and I want to change that

Fab, let's do that. Clone `https://github.com/apple/swift-nio` if you need a crutch for now?

Reply:
> Definitely an increase in faff, but we could separate the scripts out and generate the yaml files with another script (basically just to inline the scripts into the yaml). We could also shellcheck the scripts in CI + check the yaml is up-to-date.

Not a bad idea. Generating the yml from scripts isn't too bad either

Reply:
You don't even need to clone, you can just download a file from the tree, e.g. https://github.com/vapor/api-docs/blob/main/.github/workflows/build-and-deploy-docs-workflow.yml#L59-L64